projects/knora-ui/src/lib/viewer/values/text-value/text-value-as-html/text-value-as-html.component.spec.ts

```diff
@@ -0,0 +1,135 @@
+      declarations: [
+        TestHostDisplayValueComponent,
+        TextValueAsHtmlComponent
+      ],
+      imports: [
+        ReactiveFormsModule,
+        MatInputModule,
+        BrowserAnimationsModule
+      ],
+      providers: []
+    })
+      .compileComponents();
+  }));
+
+  describe('display text value with markup', () => {
+    let testHostComponent: TestHostDisplayValueComponent;
+    let testHostFixture: ComponentFixture<TestHostDisplayValueComponent>;
+    let hostCompDe;
+
+    beforeEach(() => {
+      testHostFixture = TestBed.createComponent(TestHostDisplayValueComponent);
+      testHostComponent = testHostFixture.componentInstance;
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent).toBeTruthy();
+
+      hostCompDe = testHostFixture.debugElement;
+
+    });
+
+    it('should display an existing value', () => {
+
+      const inputVal: ReadTextValueAsHtml = new ReadTextValueAsHtml();
+
+      inputVal.hasPermissions = 'CR knora-admin:Creator|M knora-admin:ProjectMember|V knora-admin:KnownUser|RV knora-admin:UnknownUser';
+      inputVal.userHasPermission = 'CR';
+      inputVal.type = 'http://api.knora.org/ontology/knora-api/v2#TextValue';
+      inputVal.id = 'http://rdfh.ch/0001/H6gBWUuJSuuO-CilHV8kQw/values/TEST_ID';
+      inputVal.html =
+        '<p>This is a <b>very</b> simple HTML document with a <a href="https://www.google.ch" target="_blank" class="kui-link">link</a></p>';
+
+      testHostComponent.displayInputVal = inputVal;
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.inputValueComponent).toBeTruthy();
+
+      const valueComponentDe = hostCompDe.query(By.directive(TextValueAsHtmlComponent));
+
+      const valueParagraph = valueComponentDe.query(By.css('div.value'));
+      const valueParagraphNativeElement = valueParagraph.nativeElement;
+
+      expect(testHostComponent.inputValueComponent.displayValue.html)
+        .toEqual('<p>This is a <b>very</b> simple HTML document with a <a href="https://www.google.ch" target="_blank" class="kui-link">link</a></p>');
+
+      expect(testHostComponent.inputValueComponent.mode).toEqual('read');
+
+      expect(valueParagraphNativeElement.innerHTML)
+        .toEqual('<p>This is a <b>very</b> simple HTML document with a <a href="https://www.google.ch" target="_blank" class="kui-link">link</a></p>');
+
+      const commentSpan = valueComponentDe.query(By.css('span.comment'));
+
+      expect(commentSpan).toBe(null);
+
+    });
+
+    it('should display an existing value with a comment', () => {
+
+      const inputVal: ReadTextValueAsHtml = new ReadTextValueAsHtml();
+
+      inputVal.type = 'http://api.knora.org/ontology/knora-api/v2#TextValue';
+      inputVal.id = 'http://rdfh.ch/0001/H6gBWUuJSuuO-CilHV8kQw/values/TEST_ID';
+      inputVal.html =
+        '<p>This is a <b>very</b> simple HTML document with a <a href="https://www.google.ch" target="_blank" class="kui-link">link and a comment</a></p>';
```

Comment:
Small change, just move the text "and a comment" outside of the anchor tag and update the expected values below

Reply:
Absolutely!

Reply:
done in https://github.com/dasch-swiss/knora-ui-ng-lib/pull/55/commits/7db19f643d18b199f0268890b92fcd14c4ba6bd3